### PR TITLE
#2532 and couple more fixes

### LINKED ISSUE #2532: Existence of Pandoc is checked locally
So in remote case if pandoc is missing *locally* we won't let user publish. On the other hand, if pandoc is missing on the server, user may get error messages in REPL.

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2532 Existence of Pandoc is checked locally

- Fix endless requests to RData for non-existing functions (semi-hang debugger data tip)
- Fix issue when opening markdown without REPL up asserts b/c VsShell hasn't been initialized as R package is not loaded yet.

Improve remote host information printing:
- Move code to components
- Print after the R information is printed

### CODE REVIEW

File: src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs [hunk 1]
```diff
@@ -27,9 +28,39 @@ public sealed class SessionInformationCommand : IAsyncCommand {
...
+
+        private void OnBrokerChanged(object sender, EventArgs e) {
+            if (_interactiveWorkflow.RSession.IsRemote) {
+                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+            }
+        }
+
+        private async Task PrintBrokerInformationAsync(bool reportTelemetry = false) {
+            var a = await _interactiveWorkflow.RSessions.Broker.GetHostInformationAsync();
+            var console = _interactiveWorkflow.RSessions.Console;
```
Comment (AlexanderSher) [inline src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs]: On the level of `InteractiveWorkflow`, we can write directly to the interactive window - no need to reference it back from `RSession`

Comment (MikhailArkhipov) [inline src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs]: OK

File: src/Host/Client/Impl/rtvs/R/util.R [hunk 2]
```diff
@@ -297,3 +297,16 @@ enable_autosave <- function(delete_existing) {
...
     });
 }
+
+exists_on_path <- function(filename) {
+    folders <- strsplit(Sys.getenv('PATH'), ';')[[1]];
+    folders <- gsub("\\", "/", folders, fixed = TRUE);
+    folders <- sub("(\\/$)", "", folders);
+    folders <- paste0(folders, "/", filename);
+    for (f in folders) {
+        if (file.exists(f)) {
```
Comment (int19h) [inline src/Host/Client/Impl/rtvs/R/util.R]: `file.exists` is vectorized, so you should be able to just do `any(file.exists(folders))`.

**Comment by int19h** [comment]
:+1:

File: src/Host/Client/Impl/rtvs/R/util.R (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Host/Client/Impl/rtvs/R/util.R]: I know but I didn't want to go through all paths and rather stop when found

File: src/Host/Client/Impl/Host/BrokerClient.cs [hunk 3]
```diff
@@ -87,6 +84,12 @@ internal abstract class BrokerClient : IBrokerClient {
             }
         }
 
+        public async Task<AboutHost> GetHostInformationAsync(CancellationToken cancellationToken) {
+            var response = await HttpClient.GetAsync("/about", cancellationToken);
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/Host/BrokerClient.cs]: Do we want to handle any exceptions here?

Comment (int19h) [inline src/Host/Client/Impl/Host/BrokerClient.cs]: This method should probably be in a separate service class, actually.

Comment (MikhailArkhipov) [inline src/Host/Client/Impl/Host/BrokerClient.cs]: Command that uses it will propagate exception to VS which typically handles and shows the error box.

Comment (MikhailArkhipov) [inline src/Host/Client/Impl/Host/BrokerClient.cs]: But we can handle typical transport/canceled if you prefer

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**